### PR TITLE
Move from travis ci to github actions

### LINKED ISSUE #1387: Move from Travis CI to GitHub actions
With the changing policies for Travis CI, we should move to GitHub actions.
A similar migration is underway across many projects in the Jupyter ecosystem (see rf. https://github.com/jupyterhub/team-compass/issues/353)

### PR DESCRIPTION
**Pull request by BertR**
In line with other Jupyter projects, we're moving the CI for nbgrader from Travis to GitHub actions.
Closes #1387

### CODE REVIEW

File: dev-requirements.txt [hunk 1]
```diff
@@ -5,7 +5,7 @@ pytest-xdist==1.34.0
 #pytest-mypy
 coverage
 selenium
-sphinx
+sphinx==3.2.1
```
Comment (jhamrick) [inline dev-requirements.txt:8]: Is pinning this required?

**Review by jhamrick — APPROVED** [review]
Thanks!

File: dev-requirements.txt (hunk 1 above)
Comment (BertR) [reply dev-requirements.txt:8]: Building the docs fails with newer versions of sphinx, but that should be a different PR to this. I will remove this change in this PR.

Comment (jhamrick) [reply dev-requirements.txt]: Sounds good, thanks!

**Pull request merged**